src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -53,14 +54,16 @@ public void CompareArtifactsToBaseline()
         {
             if (!BaselineFileContent.TryGetValue(entry.FilePath, out long baselineBytes))
             {
-                OutputHelper.LogWarningMessage($"{entry.FilePath} does not exist in baseline. Adding it to the baseline file");
+                TrackDifference($"{entry.FilePath} does not exist in baseline. Adding it to the baseline file");
             }
             else
             {
                 CompareFileSizes(entry.FilePath, entry.Bytes, baselineBytes);
```

Comment:
To avoid looping through all files twice, consider removing the filepath from the baseline content as it's encountered. That way, all the remaining baseline entries are ones that do not exist in the tarball

```
    else
    {
        CompareFileSizes(entry.FilePath, entry.Bytes, baselineBytes);
        BaselineFileContent.Remove(entry.FilePath);  // remove the entry from the baseline dictionary
    }

...

    // Now, BaselineFileContent only contains files that were in the baseline but not in tarEntries
    foreach(var entry in BaselineFileContent)
    {
        TrackDifference($"`{entry.Key}` is no longer being produced. It was {entry.Value} bytes");
    }
```